app/importers/mahonia_mapper.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+class MahoniaMapper < Darlingtonia::HashMapper
+  BEPRESS_TERMS_MAP = {
+    publication_date: :date,
+    title: :title,
+    keywords: :keyword,
+    subject: :subject,
+    document_type: :resource_type,
+    degree_name: :degree,
+    institution_name: :institution,
+    school: :school,
+    department: :department,
+    identifier: :identifier,
+    orcid: :orcid_id,
+    language: :language,
+    license: :license,
+    rights: :rights_statement,
+    note: :rights_note,
+    abstract: :description,
+    journal_title: :source
+  }.freeze
+
+  def metadata=(meta)
+    be_press_hash = meta.to_h
+    @metadata = {}
+    be_press_hash.each_key do |k|
+      if BEPRESS_TERMS_MAP.keys.include? k.to_sym
+        @metadata[BEPRESS_TERMS_MAP[k.to_sym]] = be_press_hash.delete k
+      end
+    end
+  end
+
```

Comment:
@no-reply This is what we want, and it returns a hash with symbols, so  `Array(metadata[method_name])` doesn't need to do the converting. I wonder if using method_missing in this pattern might be worth a real time talk, maybe with the rest of the team?

Reply:
We need to both map and filter the terms, actually I think I'd prefer to do this in an explicitly-named method:
  ```ruby
    @metadata = map_and_filter_bepress(meta)

    def  map_and_filter_bepress(meta)
      meta.each_key do |k|
        if BEPRESS_TERMS_MAP.keys.include? k.to_sym
          @metadata[BEPRESS_TERMS_MAP[k.to_sym]] = meta.delete k
        end
      end
    end
```

Reply:
I think this is okay, but it's also helping me understand where our miscommunication is.

`@metadata` is not really meant to store the hyrax fields, but rather the input metadata. The basic contract that a mapper is meant to fulfill is:

  - Accepts input metadata to `metadata=`; and
  - Provides a collection of symbols from `#fields`; and
  - Responds to methods matching each symbol in `#fields` to provide mapped Hyrax data.

There's no particular need to use `#method_missing`, this was just a handy way of meeting the requirement in point 3. A really simple mapper implementation might look like:

```ruby
class MyMapper
  def metadata=(meta)
    @metadata = meta
  end

  def title
     metadata['title']
  end

  def description
    metadata['description']
  end

  def date
     Date.today
  end
end
```

It's really okay to do this mapping up front and store it in metadata, but it's also less lazy; the intent is really for mappers to be able to avoid doing the mapping work until access time. In the CSV case, this is probably not a major concern.

Reply:
The storing of the metadata is in the same large step the InputRecord.from does, just before accessing, and does make the intent clearer, I think. I don't know if many implementations are going to need to do this kind of filtering and translating of the terms provided, but it's something to think about if we think they will.

Reply:
> The storing of the metadata is in the same large step the InputRecord.from does, just before accessing, and does make the intent clearer, I think.

Can you say more about this? One potential issue is that we expose `@metadata` via a reader method. In other mappers, that metadata is the exact input metadata (i.e. it's the same object, unchanged). In this mapper, the original input metadata isn't accessible.